pkgs/tools/security/gnupg/1compat.nix

```diff
@@ -4,14 +4,22 @@ stdenv.mkDerivation {
   name = "gnupg1compat-${gnupg.version}";
 
   builder = writeScript "gnupg1compat-builder" ''
+    PATH=${coreutils}/bin
     # First symlink all top-level dirs
-    ${coreutils}/bin/mkdir -p $out
-    ${coreutils}/bin/ln -s "${gnupg}/"* $out
+    mkdir -p $out
+    ln -s "${gnupg}/"* $out
 
     # Replace bin with directory and symlink it contents
-    ${coreutils}/bin/rm $out/bin
-    ${coreutils}/bin/mkdir -p $out/bin
-    ${coreutils}/bin/ln -s "${gnupg}/bin/"* $out/bin
+    rm $out/bin
+    mkdir -p $out/bin
+    ln -s "${gnupg}/bin/"* $out/bin
+
+    # Add symlinks for any executables that end in 2 and lack any non-*2 version
+    for f in $out/bin/*2; do
+      [[ -x $f ]] || continue # ignore failed globs and non-executable files
+      [[ -e ''${f%2} ]] && continue # ignore commands that already have non-*2 versions
+      ln -s -- "''${f##*/}" "''${f%2}"
+    done
```

Comment:
Do you expect other binaries to be symlinked other than `gpg` and `gpgv`?

If no, symlinking them explicitly could be a more convenient to read I think, but what you propose is valid too.

Reply:
I don't have particular reason to expect others, but the intent is rolling with whatever changes take place in the upstream packages without needing to maintain code that explicitly tests for individual binaries; if a new executable with a `2` suffix shows up in a gnupg package we could potentially be instructed to wrap in the future, with my user hat on, I would expect that to be symlinked as well.

(But then, as a counterargument, the gpg 1.x series is no longer undergoing development, so there shouldn't be new binaries added there, so the set of links potentially needed for *compatibility* is bounded, and could well be hardcoded).

So... I can see two sides here. If as a reviewer you want to direct a rewrite with more explicit code, I'm happy to do that.

Reply:
This is good for me as it is. It is just less easily readable for non bash gurus